File: Creational/Abstract Factory/Python/product/sedan_parts.py
# product/sedan_parts.py
from .auto_body_part import AutoBodyPart

class SedanLeftDoor(AutoBodyPart):
    def get_type(self) -> str:
        return "Sedan Left Door"

class SedanRightDoor(AutoBodyPart):
    def get_type(self) -> str:
        return "Sedan Right Door"

class SedanHood(AutoBodyPart):
    def get_type(self) -> str:
        return "Sedan Hood"

File: Creational/Abstract Factory/TypeScript/src/product/SedanParts.ts
// src/product/SedanParts.ts
import { AutoBodyPart } from './AutoBodyPart';

export class SedanLeftDoor implements AutoBodyPart {
    getType(): string {
        return "Sedan Left Door";
    }
}

export class SedanRightDoor implements AutoBodyPart {
    getType(): string {
        return "Sedan Right Door";
    }
}

export class SedanHood implements AutoBodyPart {
    getType(): string {
        return "Sedan Hood";
    }
}
